backend/src/process.py

```diff
@@ -538,30 +609,218 @@ def __create_node_finish(self, node_id: NodeId) -> Event:
             },
         }
 
+    def __create_node_start(self, node_id: NodeId) -> Event:
+        return {
+            "event": "node-start",
+            "data": {
+                "nodeId": node_id,
+            },
+        }
+
     def __get_output_nodes(self) -> List[NodeId]:
         output_nodes: List[NodeId] = []
         for node in self.chain.nodes.values():
             # we assume that iterator node always have side effects
-            side_effects = isinstance(node, IteratorNode) or node.has_side_effects()
+            side_effects = (
+                isinstance(node, IteratorNode)
+                or isinstance(node, NewIteratorNode)
+                or isinstance(node, CollectorNode)
+                or node.has_side_effects()
+            )
```

Comment:
Please don't assume that the new iterators and collectors have side effects and declare side effects for them just like for regular nodes. Assuming that (old) iterator nodes have side effects was just a hack to make the old system work, let's not carry it over.